src/Microsoft.Health.Dicom.Core/Features/Common/ISecretStore.cs

```diff
@@ -0,0 +1,18 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Dicom.Core.Features.Common;
+
+public interface ISecretStore
+{
+    Task DeleteSecretAsync(string secretName, CancellationToken cancellationToken = default);
+
+    Task<string> GetSecretAsync(string secretName, CancellationToken cancellationToken = default);
```

Comment:
So are secrets created per export operation? I was under the impression we were going to encrypt SAS tokens instead or storing them as secrets.

Reply:
We can do that too, but we will have to rotate the encryption key and all I think to meet the requirements.

Reply:
Mohammad's code was just storing and retrieving the secret has here

Reply:
I am just worried that if we store each SAS key individually then we'll have to manage their lifecycle (like their eventual removal). And then if the export were to fail for some reason, we'll need to remove old SAS keys... which means we'll need to probably have some timer task that runs periodically to clean up.

And at the point, perhaps we should just have a task whose job it is to rotate the encryption key every so often? Maybe there is a third approach?

Reply:
Rotation also adds the need of overlapping keys to make sure we can gracefully handle existing running ones.

Reply:
And I am not sure if it still passes/recommended in the security requirement. Meaning you can encrypt and store the secret in a normal datastore vs using a secret store